contributors/design-proposals/cluster-info-configmap.md

```diff
@@ -0,0 +1,25 @@
+# Persist cluster-wide data in ConfigMap
+
+## Abstract
+
+We want to allow cluster-wide data to be persisted for general purposes, currently for us (service catalog) is to focus on the ability to identify different clusters that access our service broker with a 'cluster-id'.
+
+## Motivation
+
+Currently there is no way for service catalog api server to identify the different clusters that access our service brokers. The goal of this design is to have a unify storage for cluster related information. The information should be retained and survive between component failures. 
+
+## Detailed design
+
+A configmap `cluster-info` is created in the `default` namespace at the starup of the controller. The initial content of the map will be as the following:
+
+```
+apiVersion: v1
+data:
+  cluster-id: uuid-of-the-cluster
```

Comment:
We alreadyhave another use-case where multiple components need to know the same ID (federation, L7 load-balancer, and controller manager).  I really want a canonical place to carry this info, and other cluster-wide flags that things sometimes need to know (e.g. CIDRs).

@nicksardo @mtaufen